### PR TITLE
api/core/v1: Add docs clarifying what's "old" core/v1 Event and what's "new" events/v1beta1 Event

### PR DESCRIPTION
**Pull request by LukeShu**
**What type of PR is this?**

/kind documentation
/sig scalability

**What this PR does / why we need it**:

With the work going in to https://github.com/kubernetes/enhancements/issues/383 / https://git.k8s.io/community/contributors/design-proposals/instrumentation/events-redesign.md it's confusing what's what.  I believe that it is important for the documentation to distinguish between pre-"events-redesign.md" and post-"events-redesign.md" things.

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:

I'm not sure (especially regarding `types.go`) that I got the style/notation "correct".

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by yastij — COMMENTED** [review]
/approve

for godoc usefulness, I'll take a deeper look later 

also

/assign @wojtek-t @liggitt

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -5046,6 +5046,10 @@ const (
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
 // Event is a report of an event somewhere in the cluster.
+//
+// There is a pending enhancement to replace `Event.v1.` with what is currently
+// `Event.v1beta1.events.k8s.io`; see <https://github.com/kubernetes/enhancements/issues/383> and
+// <https://git.k8s.io/community/contributors/design-proposals/instrumentation/events-redesign.md>.
```
Comment (wojtek-t) [inline staging/src/k8s.io/api/core/v1/types.go]: We have to migrate the proposal to KEP.
@yastij - do you have time to do that? If not I can do that next weekend when i will be flying for Kubecon.

File: staging/src/k8s.io/client-go/tools/events/doc.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
```
Comment (wojtek-t) [inline staging/src/k8s.io/client-go/tools/events/doc.go]: While the changes in types.go require API review, these ones are obviously an improvements.
If you could split those into a separate PR, we could merge it now.

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (yastij) [reply staging/src/k8s.io/api/core/v1/types.go]: sgtm, I'll take look once your PR is out

File: staging/src/k8s.io/client-go/tools/events/doc.go (hunk 2 above)
Comment (LukeShu) [reply staging/src/k8s.io/client-go/tools/events/doc.go]: Sure thing. Does it matter which PR is which?  Should the new PR be just the types.go changes, and this PR be the other changes; or vice-versa?

Comment (wojtek-t) [reply staging/src/k8s.io/client-go/tools/events/doc.go]: Doesn't matter.

Comment (LukeShu) [reply staging/src/k8s.io/client-go/tools/events/doc.go]: OK. I've moved the non-`types.go` stuff to https://github.com/kubernetes/kubernetes/pull/85134

**Comment by liggitt** [comment]
I would not expect forward-facing explanations in v1 API documentation

**Comment by liggitt** [comment]
API docs for v1 types is probably not the place to reference in-progress redesigns

/close
